cpp/test/DataStorm/api/Writer.cpp

```diff
@@ -36,8 +36,8 @@ void ::Writer::run(int argc, char* argv[])
             // Communicators shared with DataStorm must have a property set that can use the "DataStorm" opt-in prefix.
             Ice::InitializationData initData;
             initData.properties = make_shared<Ice::Properties>(vector<string>{"DataStorm"});
-            Node n2(Ice::initialize(initData));
-            n2.getCommunicator()->destroy();
+            Ice::CommunicatorHolder communicatorHolder(Ice::initialize(initData));
+            Node n2(communicatorHolder.communicator());
         }
```

Comment:
The previous code would destroy the communicator before destroying the node, which now results in CommunicatorDestroyerException thrown from DataStormI::Instance::destroy.